src/query/string_opfunc.c

```diff
@@ -4375,6 +4375,27 @@ db_string_rlike (const DB_VALUE * src, const DB_VALUE * pattern, const DB_VALUE
 	goto cleanup;
       }
 
+    INTL_CODESET src_codeset = db_get_string_codeset (src);
+    INTL_CODESET pattern_codeset = db_get_string_codeset (pattern);
+
+    if ((src_category != pattern_category) || (src_codeset != pattern_codeset))
+      {
```

Comment:
there is same condition above the source code. 'if (src category! = Pattern category)'. result is different

Reply:
i fixed it